R/linearregutils.R

```diff
@@ -74,16 +74,41 @@ processElasticNet <- function(inputs, config){
 
   # FIXME: Revisit what we pass to the weights argument.
   if (config$`Use Weight`){
-    the.model <- glmnet::glmnet(x = inputs$the.data[,var_names$x],
-      y = inputs$the.data[,var_names$y], family = "gaussian",
-      weights = inputs$the.data[,var_names$w], alpha = config$`alpha`,
-      intercept  = config$`Omit Constant`
-    )
+    if (!(config$`cv_glmnet`)) {
+      the.model <- glmnet::glmnet(x = inputs$the.data[,var_names$x],
+                                  y = inputs$the.data[,var_names$y], family = "gaussian",
+                                  weights = inputs$the.data[,var_names$w], alpha = config$`alpha`,
+                                  intercept  = config$`Omit Constant`, standardize = config$standardize_pred
+      )
+    } else {
+      the.model <- glmnet::cv.glmnet(x = inputs$the.data[,var_names$x],
+                                  y = inputs$the.data[,var_names$y], family = "gaussian",
+                                  weights = inputs$the.data[,var_names$w], alpha = config$`alpha`,
+                                  intercept  = config$`Omit Constant`, nfolds = config$`nfolds`,
+                                  standardize = config$standardize_pred
+      )
+    }
   } else {
-    the.model <- glmnet(x = inputs$the.data[,var_names$x],
-      y = inputs$the.data[,var_names$y], family = "gaussian",
-      alpha = config$`alpha`, intercept  = config$`Omit Constant`
-    )
+    if (!(config$`cv_glmnet`)) {
+      the.model <- glmnet(x = inputs$the.data[,var_names$x],
+                          y = inputs$the.data[,var_names$y], family = "gaussian",
+                          alpha = config$`alpha`, intercept  = config$`Omit Constant`,
+                          standardize = config$standardize_pred
+      )
+    } else {
+      the.model <- glmnet(x = inputs$the.data[,var_names$x],
+                          y = inputs$the.data[,var_names$y], family = "gaussian",
+                          alpha = config$`alpha`, intercept  = config$`Omit Constant`,
+                          nfolds = config$`nfolds`, standardize = config$standardize_pred
+      )
+    }
+  }
+  #Now we need to save the user's preference for the value of lambda that will be used
+  #in scoring
+  if (config$`lambda_1se`) {
+    the.model$lambda_pred <- lambda.1se
+  } else {
+    the.model$lambda_pred <- lambda.min
   }
```

Comment:
@btoomey this code can be shortened considerably by making it more DRY. You can rewrite it in 13 lines of code by exploiting some nifty language features in R.

```r
processElasticNet <- function(inputs, config){
  var_names <- getNamesFromOrdered(names(inputs$the.data), config$`Use Weight`)
  glmFun <- if (config$cv_glmnet) glmnet::cv.glmnet else glmnet::glmnet
  funParams <- list(x = as.matrix(inputs$the.data[,var_names$x]), 
    y = inputs$the.data[,var_names$y], family = 'gaussian', 
    intercept  = config$`Omit Constant`, standardize = config$standardize_pred,
    weights = if (!is.null(var_names$w)) inputs$the.data[,var_names$w] else NULL,
    nfolds = if (config$cv_glmnet) config$nfolds else NULL
  )
  the.model <- do.call(glmFun, Filter(Negate(is.null), funParams))
  the.model$lambda_pred <- config$lambda
  return(the.model)
}
```